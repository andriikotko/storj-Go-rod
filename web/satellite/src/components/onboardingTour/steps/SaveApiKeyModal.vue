// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="save-api-modal">
        <div class="save-api-modal__container">
            <OrangeExclamation/>
            <h1 class="save-api-modal__container__title">Is Your API Key Saved?</h1>
            <p class="save-api-modal__container__message">
                API Keys are only displayed once when generated. If you havenâ€™t saved your key, go back to copy and
                paste the API key to your preferred method of storing secrets (i.e. TextEdit, Keybase, etc.)
            </p>
            <div class="save-api-modal__container__buttons-area">
                <VButton
                    class="back-button"
                    width="186px"
                    height="45px"
                    label="Go Back"
                    :on-press="onBackClick"
                    :is-blue-white="true"
                />
                <VButton
                    width="186px"
                    height="45px"
                    label="Yes, it's Saved!"
                    :on-press="onConfirmClick"
                />
            </div>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import VButton from '@/components/common/VButton.vue';

import OrangeExclamation from '@/../static/images/onboardingTour/orange-exclamation.svg';

import { APP_STATE_ACTIONS } from '@/utils/constants/actionNames';

@Component({
    components: {
        OrangeExclamation,
        VButton,
    },
})

export default class SaveApiKeyModal extends Vue {
    /**
     * Toggles modal visibility.
     */
    public onBackClick(): void {
        this.$store.dispatch(APP_STATE_ACTIONS.TOGGLE_SAVE_API_KEY_MODAL);
    }

    /**
     * Proceeds to tour's last step.
     */
    public onConfirmClick(): void {
        this.$emit('confirmSave');
    }
}
</script>

<style scoped lang="scss">
    .save-api-modal {
        position: fixed;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;
        background-color: rgba(9, 21, 35, 0.85);
        display: flex;
        align-items: center;
        justify-content: center;
        font-family: 'font_regular', sans-serif;

        &__container {
            background-color: #fff;
            z-index: 1;
            padding: 35px;
            display: flex;
            flex-direction: column;
            align-items: center;
            border-radius: 6px;
            max-width: 460px;

            &__title {
                font-family: 'font_bold', sans-serif;
                font-size: 22px;
                line-height: 27px;
                color: #000;
                margin: 10px 0;
            }

            &__message {
                font-size: 16px;
                line-height: 24px;
                color: #000;
                word-break: break-word;
                text-align: center;
                margin: 0 0 10px 0;
            }

            &__buttons-area {
                display: flex;
                align-items: center;
            }
        }
    }

    .back-button {
        margin-right: 10px;
    }
</style>
