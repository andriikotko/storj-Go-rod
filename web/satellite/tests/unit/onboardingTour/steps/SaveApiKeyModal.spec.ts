// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import SaveApiKeyModal from '@/components/onboardingTour/steps/SaveApiKeyModal.vue';

import { mount } from '@vue/test-utils';

describe('SaveApiKeyModal.vue', () => {
    it('renders correctly', (): void => {
        const wrapper = mount(SaveApiKeyModal);

        expect(wrapper).toMatchSnapshot();
    });
});
