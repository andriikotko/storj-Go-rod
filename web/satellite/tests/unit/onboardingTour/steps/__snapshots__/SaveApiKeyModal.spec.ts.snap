// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveApiKeyModal.vue renders correctly 1`] = `
<div class="save-api-modal">
  <div class="save-api-modal__container"><svg width="48" height="48" viewBox="0 0 48 48" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path fill-rule="evenodd" clip-rule="evenodd" d="M0 24C1.15877e-06 10.7452 10.7452 -1.15877e-06 24 0C37.2548 2.09815e-06 48 10.7452 48 24C48 37.2548 37.2548 48 24 48C10.7452 48 -2.09815e-06 37.2548 0 24ZM26.6669 24.0003C26.6669 25.473 25.473 26.6669 24.0003 26.6669C22.5275 26.6669 21.3336 25.473 21.3336 24.0003L21.3336 13.3336C21.3336 11.8609 22.5275 10.6669 24.0003 10.6669C25.473 10.6669 26.6669 11.8609 26.6669 13.3336L26.6669 24.0003ZM24.0003 37.3335C25.473 37.3335 26.6669 36.1396 26.6669 34.6668C26.6669 33.194 25.473 32.0001 24.0003 32.0001C22.5275 32.0001 21.3336 33.194 21.3336 34.6668C21.3336 36.1396 22.5275 37.3335 24.0003 37.3335Z" fill="#F4B000"></path>
    </svg>
    <h1 class="save-api-modal__container__title">Is Your API Key Saved?</h1>
    <p class="save-api-modal__container__message">
      API Keys are only displayed once when generated. If you havenâ€™t saved your key, go back to copy and
      paste the API key to your preferred method of storing secrets (i.e. TextEdit, Keybase, etc.)
    </p>
    <div class="save-api-modal__container__buttons-area">
      <div class="back-button container blue-white" style="width: 186px; height: 45px;"><span class="label">Go Back</span></div>
      <div class="container" style="width: 186px; height: 45px;"><span class="label">Yes, it's Saved!</span></div>
    </div>
  </div>
</div>
`;
